Test/git-issues/git-issue-1604f.dfy.expect

```diff
@@ -0,0 +1,20 @@
+git-issue-1604f.dfy(38,8): Error: Could not prove that the bound variable 'f' satisfies the (non runtime testable) type SpecialFruit after the range, where it's assumed to be of type Fruit based on inferred bounds.
+git-issue-1604f.dfy(16,31): [Related location] The constraint of SpecialFruit cannot be tested at run-time because a call to a ghost predicate is allowed only in specification contexts (consider declaring the predicate with 'predicate method')
+git-issue-1604f.dfy(38,40): Error: possible violation of function precondition. Careful: variable f has type Fruit and not SpecialFruit because the range is compiled and SpecialFruit cannot be tested at run-time
```

Comment:
So, why don't we get an error about a precondition violation?

Reply:
`git-issue-1604f.dfy(38,40): Error: possible violation of function precondition`
is an error about a precondition violation. I'm just adding more useful information in case some users are confused, as @RustanLeino  suggested. What do you mean here?

Reply:
I don't understand why the original precondition violation error is not shown. Based on the code it seemed like you were adding an extra error but not hiding an existing now.

Reply:
The code shows I'm adding an extra message to the original precondition violation error:
https://github.com/dafny-lang/dafny/pull/1702/files#diff-2375e053597753dae383e30da45bce3a97e5fa2e7e9a3fc601cf68923df03516R5747
error message used here (unchanged lines)
https://github.com/dafny-lang/dafny/pull/1702/files#diff-2375e053597753dae383e30da45bce3a97e5fa2e7e9a3fc601cf68923df03516R5758-R5760

Reply:
I guess you're overwriting the error message from line 5737:
`string errorMessage = CustomErrorMessage(p.Attributes);`

Reply:
Only if `errorMessage == null`, so I wouldn't call it "overriding", just "provide another default message".
But you pointed out one important thing: There was duplication in the error, which can now be resolved by the new ProofObligationDescription. I added an optional hint to it and I'm now only filling that hint instead of modifying the error message.